src/geometamaker/geometamaker.py

```diff
@@ -34,13 +34,15 @@ def detect_file_type(filepath):
         return 'table'
     if desc.compression:
         return 'archive'
-    gis_type = pygeoprocessing.get_gis_type(filepath)
+    try:
+        gis_type = pygeoprocessing.get_gis_type(filepath)
+    except ValueError:
+        raise ValueError(
+            f'{filepath} does not appear to be one of (archive, table, raster, vector)')
     if gis_type == pygeoprocessing.VECTOR_TYPE:
         return 'vector'
     if gis_type == pygeoprocessing.RASTER_TYPE:
         return 'raster'
```

Comment:
What happens when we have a dataset that is both a raster and a vector (e.g. a PDF)?

Pygeoprocessing's GIS type is a bitwise mask, so it's very possible that our value would fall through the conditional here and the function would return `None` (which I think would be unexpected).

Reply:
> What happens when we have a dataset that is both a raster and a vector

Oh great question. We should figure out what we _want_ to happen in this case, not just from the perspective of error-handling.

Reply:
@phargogh , I added another exception for datasets that are both raster and vector. I could imagine supporting these, but I'd rather wait until we hear from someone that needs this support. 

I managed to create a GPKG in QGIS that contains both a raster layer and a vector layer and confirmed `pygeoprocessing` correctly returns the type as `3`. But I don't know how to create such a file using GDAL since the drivers seem to be either raster or vector. And I don't think it's worth the effort to build a test case.